implementations/java-json-schema/BowtieJsonSchema.java

```diff
@@ -115,18 +138,55 @@ private void dialect(JsonNode node) throws JsonProcessingException {
       DialectRequest.class
     );
 
-    // well, setting default dialect shouldn't be necessary but why not
-    if (SpecificationVersion.DRAFT2020_12.getId().equals(dialectRequest.dialect())) {
-        validatorFactory.withDefaultDialect(new Dialects.Draft2020Dialect());
-    } else if (SpecificationVersion.DRAFT2019_09.getId().equals(dialectRequest.dialect())) {
-        validatorFactory.withDefaultDialect(new Dialects.Draft2019Dialect());
-    } else if (SpecificationVersion.DRAFT7.getId().equals(dialectRequest.dialect())) {
-      validatorFactory.withDefaultDialect(new Dialects.Draft7Dialect());
+    try {
+      if (getSpecificationVersionFor("DRAFT2020_12").getId().equals(dialectRequest.dialect())) {
+        setDialect(Class.forName("dev.harrel.jsonschema.Dialects$Draft2020Dialect"));
+      } else if (getSpecificationVersionFor("DRAFT2019_09").getId().equals(dialectRequest.dialect())) {
+        setDialect(Class.forName("dev.harrel.jsonschema.Dialects$Draft2019Dialect"));
+      } else if (getSpecificationVersionFor("DRAFT7").getId().equals(dialectRequest.dialect())) {
+        setDialect(Class.forName("dev.harrel.jsonschema.Dialects$Draft7Dialect"));
+      }
+    } catch (Exception e) {
+      throw new IllegalStateException("Failed to set dialect", e);
     }
+
     DialectResponse dialectResponse = new DialectResponse(true);
     output.println(objectMapper.writeValueAsString(dialectResponse));
   }
 
+  private void setDialect(Class<?> dialectClass) throws Exception {
+    Object dialectInstance = dialectClass.getDeclaredConstructor().newInstance();
```

Comment:
Maybe instead of calling `getDeclaredConstructor()` (which can return also non-public constructors and run into some security/access issues) you could call `getConstructor()` (which returns only public ctors, and these classes should have public ctors)